File: 채주혁/1주차/programmers_정수삼각형.java
class Solution {
    public int solution(int[][] triangle) {
        int answer = 0;
        int n = triangle.length;
        int[][] dp = new int[n+1][n+1];

        dp[1][1] = triangle[0][0];
        for(int i=2; i<=triangle.length; i++) {
            for(int j=1; j<=i; j++) {
                if(j==1) {
                    dp[i][j] = dp[i-1][j] + triangle[i-1][j-1];
                }
                else if(j==i) {
                    dp[i][j] = dp[i-1][j-1] + triangle[i-1][j-1];
                }
                else {
                    dp[i][j] = Math.max(dp[i-1][j-1],dp[i-1][j])+triangle[i-1][j-1];
                }
            }
        }
        for(int i=1; i<=n; i++) {
            answer = Math.max(answer,dp[n][i]);
        }

        return answer;
    }
}

File: 신혜원/week2/programmers_정수 삼각형.js
function solution(triangle) {
    // 깊은 복사
    let dp = triangle.map(row => [...row]);

    // 아래에서부터 올라가며 최대 경로 합 계산
    for (let i = dp.length - 2; i >= 0; i--) {
        for (let j = 0; j < dp[i].length; j++) {
            dp[i][j] += Math.max(
                dp[i + 1][j],
                dp[i + 1][j + 1]
            );
        }
    }

    return dp[0][0];
}